entity-framework/core/modeling/custom-function-mapping.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Custom function mapping - EF Core
+description: Mapping user-defined functions to database functions
+author: maumar
+ms.date: 11/23/2020
+uid: core/modeling/custom-function-mapping
+---
+# Custom function mapping
+
+EF Core allows for using user-defined SQL functions in queries. To do that, the functions need to be mapped to a CLR method during model configuration. When translating the LINQ query to SQL, user-defined function will be called instead of the CLR function it has been mapped to.
+
+## Mapping method to a SQL function
+
+To illustrate the custom function mapping, lets define the following entities:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Blog.cs#Entity)]
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Post.cs#Entity)]
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Tag.cs#Entity)]
+
+And the following model configuration:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/BloggingContext.cs#EntityConfiguration)]
```

Comment:
Do we really need many to many model for this example? I prefer it to be simpler example which works with minimal model configuration for ease of understanding of this particular topic.

Reply:
I had the same thought, just to keep samples as dead-simple as possible.

Reply:
Ok will Make them comments instead, so 2x 1-many